Test/Microsoft.ApplicationInsights.Test/Shared/Extensibility/Implementation/Tracing/HeartbeatTests.cs

```diff
@@ -0,0 +1,415 @@
+                hbeat.Initialize(configuration: null);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatTwiceDoesntFail()
+        {
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: null);
+                hbeat.Initialize(configuration: null);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatDefaultsAreSetProperly()
+        {
+            using (var hbeat = new HealthHeartbeatProviderMock())
+            {
+                hbeat.Initialize(configuration: null);
+                Assert.IsNull(hbeat.DisabledHeartbeatProperties);
+                Assert.AreEqual(HeartbeatProvider.DefaultHeartbeatIntervalMs, hbeat.HeartbeatInterval.TotalMilliseconds);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatWithNonDefaultInterval()
+        {
+            TimeSpan nonDefaultInterval = TimeSpan.FromMilliseconds(10000);
+
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: null, timeBetweenHeartbeats: nonDefaultInterval);
+                Assert.AreEqual(nonDefaultInterval, hbeat.HeartbeatInterval);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatWithNullFieldsFails()
+        {
+            using (var hbeat = new HeartbeatProvider())
+            {
+                bool initResult = hbeat.Initialize(configuration: null, timeBetweenHeartbeats: TimeSpan.FromMilliseconds(10000), disabledDefaultFields: null);
+                Assert.IsTrue(initResult, "Initialization without allowed dissallowed fields should be fine.");
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatWithZeroIntervalFails()
+        {
+            using (var hbeat = new HeartbeatProvider())
+            {
+                bool initResult = hbeat.Initialize(configuration: null, timeBetweenHeartbeats: TimeSpan.FromMilliseconds(0), disabledDefaultFields: null);
+                Assert.IsFalse(initResult, "Initialization without a valid delay value (0) should fail.");
+            }
+        }
+
+        [TestMethod]
+        public void CanExtendHeartbeatPayload()
+        {
+
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: new TelemetryConfiguration());
+
+                try
+                {
+                    Assert.IsTrue(hbeat.AddHealthProperty("test01", "this is a value", true));
+                }
+                catch (Exception e)
+                {
+                    Assert.Fail(string.Format(CultureInfo.CurrentCulture, "Registration of a heartbeat payload provider throws exception '{0}", e.ToInvariantString()));
+                }
+            }
+        }
+
+        [TestMethod]
+        public void CanSetDelayBetweenHeartbeats()
```

Comment:
Sine HeartBeatProvider is internal, i assume user would never set this directly. So, the test should be validating that setting HeartBeatInterval on DiagnosticTelemetryModule will set the same on HeartBeatProvider - right?

Reply:
Agreed, done (see below)